### PR TITLE
Added the first test.

### PR DESCRIPTION
**Pull request by crdoconnor**

### CODE REVIEW

File: turkey/templates/_base.html [hunk 1]
```diff
@@ -35,7 +35,7 @@
 <!-- Register on login page -->
 <!-- Forgot username/password on login page -->
 <!-- Allow email login on login page -->
-          <p class="navbar-text">Hi, you can <a href="{{ url_for('register') }}">register</a> or </p>
+          <p class="navbar-text">Hi, you can <a id="register-link" href="{{ url_for('register') }}">register</a> or </p>
```
Comment (geokala) [inline turkey/templates/_base.html]: The merge conflict is on this line- I was building in registration blocking capabilities while I was travelling (sorry!).

**Comment by geokala** [comment]
My first attempt to run the tests (from the root turkey directory with: hitch test tests) seems to have resulted in failure- I'm guessing because you were running your tests from the tests subfolder.

Can you note in the README where you expect the tests to be run from?

Also, if I were to want to integrate this with Travis how well does hitch co-operate (I've not dug into the docs so I'm happy if the answer is RTFM)?

**Comment by geokala** [comment]
Oh, I also made an example turkey.conf file that can be copied.

**Comment by geokala** [comment]
Also, if ~/.turkey.db doesn't exist when starting the test (from the tests directory with hitch test .), it fails:
--> 101 :         remove(path.join(expanduser("~"), ".turkey.db"))
    102 :         check_call([self.python_package.python, "manage.py", "db", "upgrade"])
    103 :         
    


builtins.FileNotFoundError

**Comment by crdoconnor** [comment]
* Yep, run the tests from the tests subfolder. I'll amend the readme.
* It runs on travis. It takes a long time to run on travis however (20 minutes) because it can't cache the stuff it downloads but it does run. 
* The last error was my stupid fault. There should be a line checking if turkey.db exists before trying to delete it:

    if path.exists(path.join(expanduser("~"), ".turkey.db")):
        remove(path.join(expanduser("~"), ".turkey.db"))

**Comment by crdoconnor** [comment]
The conflict is fixed now.

I've got a test running on travis-ci now (there's a .travis.yml), although it doesn't like having the sqlite database in the home directory so it's still failing:

https://travis-ci.org/crdoconnor/turkey/builds/104447218#L3742

If you can tell me how to change the location I can get it to start passing.

**Comment by crdoconnor** [comment]
Ah, it's failing because of the new conf file. Will fix that in a bit.

**Comment by geokala** [comment]
Yeah, it would need to generate a conf file with whatever path it wants. It has to use absolute because otherwise it creates it in one directory (relative to the manage.py) and then tries to read from it in another (relative to the turkey module).

I'm aiming to look at this again tomorrow. Is there a way to tell hitch not to download postgres (and anything else except python and the requirements)?

**Comment by crdoconnor** [comment]
Fixed.

it now writes out a turkey.conf file itself and it doesn't download or run postgres (I commented those lines out).

**Comment by crdoconnor** [comment]
Travis is now happy too: https://travis-ci.org/crdoconnor/turkey/

**Comment by geokala** [comment]
Looks good, and gives me an incentive to set up an account on Travis. Thanks!

**Pull request merged**